catascript/catascript.py

```diff
@@ -0,0 +1,217 @@
+    for id_mission_name in list_of_ids:
+        if catalog_line[id_mission_name] != None:
+            flag = flag or True
+            found_ids[id_mission_name] = catalog_line[id_mission_name]
+    
+    #Results
+    return((flag, found_ids))
+
+#Database relation functions
+def initialize_database():
+    """
+    This function initializes a database with SQL Alchemy ; its model is located in catascript.base.Catalog
+    """
+    #instantiates database
+    Base.metadata.create_all(engine)
+
+def add_entry_to_database(value_fields_dict):
+    """
+    This function adds a new value to the database. 
+    It matches the list of all possible fields given with the keys of the value of fields dictionnary, 
+    filling with None if there is not such value for that TIC.
+
+    :param value_fields_dict: dictionary containing the value for each of the needed fields for that TIC entry.
+    :type value_fields_dict: dict
+    """
+    session = Session()
+    try:
+        new_entry = Catalog(value_fields_dict)
+        session.add(new_entry)
+        session.commit()
+    except (IntegrityError, UniqueViolation):
+        #There is already an entry in the database
+        pass
+    session.close
+
+#Main function
+def catascript():
+    """
+    Builds a database containing the TESS ID (TIC), IDs for other missions, ra and dec value.
+    It then fills it with the catalog entries that have a light curve (found in a previously extracted dictionnary) and other mission ID.
+    """
+    print("Launching : catascript")
+    #load environment variables
+    load_dotenv()
+
+    #get catalog files
+    catalog_files_list = get_catalog_files()
+
+    #extract TICS_dict
+    print("Loading : TICS dict ... ", end='')
+    TICS_dict = load_TICS_dict()
+    print("Done.")
+
+    #inialize database
+    print("Initializing : database ... ", end='')
+    initialize_database()
+    print("Done.")
+
+    #get re_launch boolean
+    need_re_launch = int(os.getenv("RE_LAUNCH"))
+
+    #check if a recompute is being asked
+    if need_re_launch:
+        #loop over catalog lines in catalog
+        for catalog_file in catalog_files_list:
+            with open(catalog_file, newline='') as catalog_csv:
+                catalog_reader = csv.reader(catalog_csv, delimiter=',', quotechar='|')
+                print("Processing : catalog {} ... ".format(catalog_file), end='')
+                
+                #Iteration on each line of the csv file
+                counter = 5
+                for catalog_line in catalog_reader:
+                    #preprocess that line to put in a good format
+                    catalog_line_values = preprocess_catalog_line(catalog_line)
+                    TIC_in_catalog = catalog_line_values["ID"]
+
+                    #check if match with a TIC with known light curve
+                    (TIC_in_dict, dict_values) = check_in_TICS_dict(TIC_in_catalog, TICS_dict)
+                    if TIC_in_dict:
```

Comment:
Is it possible to display/store statistics about the number of TIC from the catalog found in the light curve's folder?

Reply:
I'm gonna make this an issue :)

Reply:
Well, just to be sure, you want to add a stat that indicates the number of lightcurves fund for a particular ID ? Or just the total number of entries put in the database.

If it's the later, I don't think that would be a good idea, as we're dealing with a normalized database.

I think it would be better to define a function somewhere else to make this query on the database

Reply:
I'm not talking about adding complexity to the database. My idea is just to display something like `X objects from the catalog were found to have light curves from TESS`

Reply:
Will be dealt in #9 :)